source/slang/slang-ir-spirv-legalize.cpp

```diff
@@ -247,6 +247,11 @@ struct SPIRVLegalizationContext : public SourceEmitterBase
             }
             else if(const auto spirvAsmOperand = as<IRSPIRVAsmOperandInst>(user))
             {
+                // Skip load's for referenced `Input` variables since a ref implies
+                // passing as is, which needs to be a pointer (pass as is).
+                if (user->getDataType()->getOp() == kIROp_RefType
+                    && storageClass == SpvStorageClassInput)
+                    return;
```

Comment:
Shouldn't this be continue or break rather than return?

Reply:
it should be `continue`

I will add to the test cases as well